core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -402,6 +403,7 @@ class TransactionCoordinator(brokerId: Int,
 
               preSendResult match {
                 case Left(err) =>
+                  info(s"TransactionalId $transactionalId failing transition to $newMetadata due to $err")
```

Comment:
I think we can remove the info log at line 393 above as we are printing all cases here anyways.

PS, may be we can update the log entry to include both the coordinator epoch as well? 

```
info(s"Updating $transactionalId's transaction state to $newMetadata with coordinator epoch $coordinatorEpoch for $transactionalId failed since the transaction coordinator epoch " +
                      s"has been changed to ${epochAndMetadata.coordinatorEpoch} after the transaction metadata has been successfully appended to the log")
```